modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/APMMetric.java

```diff
@@ -0,0 +1,133 @@
+    private AtomicReference<APMServices> services = new AtomicReference<>();
+
+    record APMServices(Meter meter, OpenTelemetry openTelemetry) {}
+
+    // TODO remove duplication between APMTracer and APMMetric. enabled, create apm services etc
+    public APMMetric(Settings settings) {
+        this.enabled = APM_ENABLED_SETTING.get(settings);
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        if (enabled) {
+            createApmServices();
+        } else {
+            destroyApmServices();
+        }
+    }
+
+    @Override
+    protected void doStart() {
+        if (enabled) {
+            createApmServices();
+        }
+    }
+
+    @Override
+    protected void doStop() {
+        destroyApmServices();
+    }
+
+    @Override
+    protected void doClose() {}
+
+    @Override
+    public <T> DoubleCounter registerDoubleCounter(MetricName name, String description, T unit) {
+        return instruments.registerDoubleCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleUpDownCounter registerDoubleUpDownCounter(MetricName name, String description, T unit) {
+        return instruments.registerDoubleUpDownCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleGauge registerDoubleGauge(MetricName name, String description, T unit) {
+        return instruments.registerDoubleGauge(name, description, unit);
+    }
+
+    @Override
+    public <T> DoubleHistogram registerDoubleHistogram(MetricName name, String description, T unit) {
+        return instruments.registerDoubleHistogram(name, description, unit);
+    }
+
+    @Override
+    public <T> LongCounter registerLongCounter(MetricName name, String description, T unit) {
+        return instruments.registerLongCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> LongUpDownCounter registerLongUpDownCounter(MetricName name, String description, T unit) {
+        return instruments.registerLongUpDownCounter(name, description, unit);
+    }
+
+    @Override
+    public <T> LongGauge registerLongGauge(MetricName name, String description, T unit) {
+        return instruments.registerLongGauge(name, description, unit);
+    }
+
+    @Override
+    public <T> LongHistogram registerLongHistogram(MetricName name, String description, T unit) {
+        return instruments.registerLongHistogram(name, description, unit);
+    }
+
+    void createApmServices() {
+        assert this.enabled;
+        assert this.services.get() == null;
+
+        AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
+            var openTelemetry = GlobalOpenTelemetry.get();
+            var meter = openTelemetry.getMeter("elasticsearch");
+
+            this.services.set(new APMServices(meter, openTelemetry));
+            instruments.setProvider(meter);
+
+            return null;
+        });
+    }
+
+    private void destroyApmServices() {
+        instruments.setProvider(null);
```

Comment:
here the alternative might be to just call setProvider(OpenTelemetry.noop().getMeter("elasticsearch"))
with that we would not need a null check in `instrument.set(meter != null ? buildInstrument(meter) : getNOOP());` and it would just always be `instrument.set(buildInstrument(meter))`
wdyt?